backend/groth16/bls12-377/mpcsetup/phase2.go

```diff
@@ -129,125 +285,93 @@ func InitPhase2(r1cs *cs.R1CS, srs1 *Phase1) (Phase2, Phase2Evaluations) {
 
 	// Prepare default contribution
 	_, _, g1, g2 := curve.Generators()
-	c2.Parameters.G1.Delta = g1
-	c2.Parameters.G2.Delta = g2
+	p.Parameters.G1.Delta = g1
+	p.Parameters.G2.Delta = g2
 
 	// Build Z in PK as τⁱ(τⁿ - 1)  = τ⁽ⁱ⁺ⁿ⁾ - τⁱ  for i ∈ [0, n-2]
 	// τⁱ(τⁿ - 1)  = τ⁽ⁱ⁺ⁿ⁾ - τⁱ  for i ∈ [0, n-2]
-	n := len(srs.G1.AlphaTau)
-	c2.Parameters.G1.Z = make([]curve.G1Affine, n)
-	for i := 0; i < n-1; i++ {
-		c2.Parameters.G1.Z[i].Sub(&srs.G1.Tau[i+n], &srs.G1.Tau[i])
-	}
-	bitReverse(c2.Parameters.G1.Z)
-	c2.Parameters.G1.Z = c2.Parameters.G1.Z[:n-1]
-
-	// Evaluate L
-	nPrivate := internal + secret
-	c2.Parameters.G1.L = make([]curve.G1Affine, nPrivate)
-	evals.G1.VKK = make([]curve.G1Affine, public)
-	offset := public
-	for i := 0; i < nWires; i++ {
-		var tmp curve.G1Affine
-		tmp.Add(&bA[i], &aB[i])
-		tmp.Add(&tmp, &C[i])
-		if i < public {
-			evals.G1.VKK[i].Set(&tmp)
-		} else {
-			c2.Parameters.G1.L[i-offset].Set(&tmp)
-		}
+	n := len(commons.G1.AlphaTau)
```

Comment:
We already initialized variable `size := len(commons.G1.AlphaTau)`. Could reuse here.